script/gen_ca.sh

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
+openssl ecparam -name secp521r1 -genkey -noout -out dummy_upstream_ca.key
+openssl req -new -x509 -key dummy_upstream_ca.key -out dummy_upstream_ca.crt -days 1825 -subj "/C=US/ST=/L=/O=SPIFFE/OU=/CN=/"  -config <(
+cat <<-EOF
+[req]
+default_bits = 2048
+default_md = sha512
+distinguished_name = dn
+[ dn ]
+[alt_names]
+URI.1 = spiffe://local
+[v3_req]
+subjectKeyIdentifier=hash
```

Comment:
Now, having the Subject Key Identifier, `openssl verify` fails (if `-partial_chain` is not specified), even with the subject `"/C=US/ST=/L=/O=SPIFFE/OU=/CN=/"`. I guess this is exactly what we expect to happen.

Reply:
Hmm this is a little bit of a surprise to me, but I guess it makes sense... if SKI/AKI is not there subject is all it has to build path with, but if it is there then it's obvious that the chain continues... It makes me a little more hesitant to introduce this, though given that SKI is a MUST, I feel we really should...

... perhaps we need to bump the priority on the issue resulting from #306. It will be problematic to lose compat with openssl and its derivatives